adapters/discord/src/bot.ts

```diff
@@ -25,6 +27,34 @@ export class DiscordBot extends Bot<DiscordBot.Config> {
     ctx.plugin(WsClient, this)
   }
 
+  private async _ensureWebhook(channelId: string) {
+    let webhook: Webhook;
+    let webhooks = await this.internal.getChannelWebhooks(channelId)
+    let selfId = this.selfId
+    if (!webhooks.find(v => v.name === "Koishi" && v.user.id === selfId)) {
+      webhook = await this.internal.createWebhook(channelId, {
+        name: "Koishi"
+      })
+      // webhook may be `AxiosError: Request failed with status code 429` error
+    } else {
+      webhook = webhooks.find(v => v.name === "Koishi" && v.user.id === this.selfId)
+    }
+    return this.webhooks[channelId] = webhook
```

Comment:
bot.webhooks 只有写入没有读取，应该是无效代码。

```suggestion
    return webhook
```